src/domain/individualHarbor/forms/pierForm/PierEditForm.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useMutation, useQuery } from '@apollo/react-hooks';
+
+import { FormProps, Pier } from '../types';
+import PierForm from './PierForm';
+import { DELETE_PIER_MUTATION, UPDATE_PIER_MUTATION } from './mutations';
+import {
+  UPDATE_PIER,
+  UPDATE_PIERVariables as UPDATE_PIER_VARS,
+} from './__generated__/UPDATE_PIER';
+import { PIER_AND_BOAT_TYPES_QUERY } from './queries';
+import LoadingSpinner from '../../../../common/spinner/LoadingSpinner';
+import { getBoatTypes, getPier } from './utils/utils';
+import { PIER_AND_BOAT_TYPES } from './__generated__/PIER_AND_BOAT_TYPES';
+import {
+  DELETE_PIER,
+  DELETE_PIERVariables as DELETE_PIER_VARS,
+} from './__generated__/DELETE_PIER';
+
+interface Props
+  extends Omit<FormProps<Pier>, 'initialValues' | 'isSubmitting'> {
+  pierId: string;
+}
+
+const PierEditForm: React.FC<Props> = ({
+  pierId,
+  onCancel,
+  onSubmit,
+  onDelete,
+  refetchQueries,
+}) => {
+  const { loading, error, data } = useQuery<PIER_AND_BOAT_TYPES>(
+    PIER_AND_BOAT_TYPES_QUERY,
+    { variables: { id: pierId } }
+  );
+  const [updatePier, { loading: isSubmitting }] = useMutation<
+    UPDATE_PIER,
+    UPDATE_PIER_VARS
+  >(UPDATE_PIER_MUTATION, {
+    refetchQueries: [
+      ...(refetchQueries ?? []),
+      { query: PIER_AND_BOAT_TYPES_QUERY, variables: { id: pierId } },
+    ],
+  });
+  const [deletePier] = useMutation<DELETE_PIER, DELETE_PIER_VARS>(
+    DELETE_PIER_MUTATION,
+    {
+      refetchQueries: refetchQueries ?? [],
+    }
+  );
+  const { t } = useTranslation();
+
+  if (loading) return <LoadingSpinner isLoading={loading} />;
+  if (error) return <div>{t('forms.common.error')}</div>;
```

Comment:
Should we take into account mutations' errors?

Reply:
Added the mutations' errors to this "catch-all" if. Could create a separate issue for creating a more descriptive error component for form errors?

Reply:
Great! Ville might have something in mind regarding these errors.